[Merged by Bors] - Add Run a Node guide

kevinbogner opened the pull request:
## Issue Addressed

Related to #3672  

## Proposed Changes

- Added a guide to run a node. Mainly, copy and paste from 'Merge Migration' and 'Checkpoint Sync'.
- Ranked it high in ToC:
  - Introduction
  - Installation
  - Run a Node
  - Become a Validator
	...
- Hid 'Merge Migration' in ToC.

## Additional Info

- Should I add/rephrase/delete something?
- Now there is some redundancy:
  - 'Run a node' and 'Checkpoint Sync' contain similar information.
  - Same for 'Run a node' and 'Become a Validator'.

kevinbogner committed:
add run_a_node guide

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/sigp/lighthouse?pullRequest=3681) <br/>All committers have signed the CLA.

michaelsproul commented:
github was being super flaky all day so I didn't get a chance to review this yet

will get onto it soon!

michaelsproul commented on book/src/run_a_node.md:
```diff
@@ -0,0 +1,145 @@
+see the [FAQ](#faq) for further information about why many:1 and 1:many configurations are not
+supported.
+
+### FAQ
```
Rather than duplicating the FAQ I think we can link to the FAQs on the merge migration and checkpoint sync pages

michaelsproul commented on book/src/run_a_node.md:
```diff
@@ -0,0 +1,145 @@
+Checkpoint sync is *optional*; however,  we recommend it since it is substantially faster
+than syncing from genesis while still providing all the same features.
+
+## Step 1: Checkpoint sync
```
I think we could break this down into more granular steps like:

1. Choose a checkpoint sync provider (describe just how to choose the URL)
2. Set up an execution node (link to EL docs, etc)
3. Run Lighthouse
4. Check logs (example of `INFO Starting checkpoint sync` and `INFO Synced`)
5. Further reading (links to other sections like Becoming a Validator, Validator Monitor, Checkpoint Sync, Merge-Migration, HTTP API, etc).

In step 3 we show an example of the full command-line used for a staking node and a non-staking node respectively, e.g.

Staking:

```bash
lighthouse bn \
  --http \
  --checkpoint-sync-url https://mainnet.checkpoint.sigp.io \
  --execution-endpoint http://localhost:8551 \
  --execution-jwt /secrets/jwt.hex 
```

Non-staking:

```bash
lighthouse bn \
  --checkpoint-sync-url https://mainnet.checkpoint.sigp.io \
  --execution-endpoint http://localhost:8551 \
  --execution-jwt /secrets/jwt.hex \
  --disable-deposit-contract-sync
```

michaelsproul reviewed (commented):
This is looking structurally very nice!

I had a bit more of a radical overhaul in mind, if you have time to implement some of those changes that would be awesome. It doesn't need to be 100% perfect though, and we can always refine it incrementally after merging.

Thanks again!

michaelsproul commented:
Oh, would also be cool to mention the `--network` flag so users can choose a network.

kevinbogner commented:
A radical overhaul sounds excellent haha. 

I will probably be done early next week. I will ping you on dc to let you know it's ready for review

kevinbogner committed:
added some steps for 'run a node'

kevinbogner committed:
add step 1 - step 3

kevinbogner committed:
step 4+move 'merge migration' to 'advanced usage'

kevinbogner committed:
step 5

kevinbogner committed:
minor changes to wording and typos

kevinbogner commented on book/src/SUMMARY.md:
```diff
@@ -46,6 +46,7 @@
     * [Pre-Releases](./advanced-pre-releases.md)
     * [Release Candidates](./advanced-release-candidates.md)
     * [MEV and Lighthouse](./builders.md)
+    * [Merge Migration](./merge-migration.md)
```
Added 'Merge Migration' to the 'Advanced Usage' section. Not sure if this is the right place for it. What do you think?

michaelsproul replied:
I think that's good!

kevinbogner commented on book/src/run_a_node.md:
```diff
@@ -0,0 +1,169 @@
+
+Once Lighthouse runs, we can monitor the logs to see if it is syncing correctly.
+
+## Step 4: Check logs
```
Is the 'Check logs' information I provide enough? Is there anything I could add or delete?
I'm struggling the most to apply appropriate information for this section

michaelsproul replied:
I think it's sufficient!

kevinbogner commented on book/src/run_a_node.md:
```diff
@@ -0,0 +1,169 @@
+
+- `lighthouse` (no flag): Mainnet.
+- `lighthouse --network mainnet`: Mainnet.
+- `lighthouse --network prater`: Prater (testnet).
```
Is 'Prater' still the current testnet?

michaelsproul replied:
Kind of. It merged with Goerli and we support both names as an alias for the same thing. I think in these user-facing docs we may as well call it Goerli and link to the Goerli homepage: https://goerli.net/

The other long-lived testnet is Sepolia, which I think we should list too (https://sepolia.dev/).

And for completeness, Gnosis chain (https://www.gnosis.io/), which is a side-chain using Ethereum consensus.

kevinbogner committed:
add merge migration

kevinbogner reviewed (commented):
Added your suggested steps.

michaelsproul commented on book/src/run_a_node.md:
```diff
@@ -0,0 +1,169 @@
+1. [Check logs](#step-4-check-logs); and
+1. [Further readings](#step-5-further-readings).
+
+Checkpoint sync is *optional*; however,  we recommend it since it is substantially faster
```
```suggestion
Checkpoint sync is *optional*; however, we recommend it since it is substantially faster
```

michaelsproul commented on book/src/run_a_node.md:
```diff
@@ -0,0 +1,169 @@
+
+## Step 1: Choose a checkpoint sync provider
+
+Since version 2.0.0 Lighthouse supports syncing from a recent finalized checkpoint.
```
```suggestion
Lighthouse supports fast sync from a recent finalized checkpoint.
```

michaelsproul commented on book/src/run_a_node.md:
```diff
@@ -0,0 +1,169 @@
+
+The Ethereum community provides various [public endpoints](https://eth-clients.github.io/checkpoint-sync-endpoints/) for you to choose from for your initial checkpoint state. Select one for your network and use it as the URL.
+
+For example, the URL for the checkpoint sync of the mainnet from Sigma Prime is `https://mainnet.checkpoint.sigp.io`,
```
```suggestion
For example, the URL for Sigma Prime's checkpoint sync server for mainnet is `https://mainnet.checkpoint.sigp.io`,
```

michaelsproul commented on book/src/run_a_node.md:
```diff
@@ -0,0 +1,169 @@
+
+In [step 3](#step-3-run-lighthouse), when running Lighthouse,
+we will enable checkpoint sync by providing the URL to the `--checkpoint-sync-url` flag.
+For now, we will copy the URL to the clipboard, selected in one of the following ways.
```
```suggestion
For now, we will copy the URL to the clipboard.
```

I think lets streamline this to recommend using a community endpoint, and add an aside about syncing from an existing node.

michaelsproul commented on book/src/run_a_node.md:
```diff
@@ -0,0 +1,169 @@
+we will enable checkpoint sync by providing the URL to the `--checkpoint-sync-url` flag.
+For now, we will copy the URL to the clipboard, selected in one of the following ways.
+
+### Automatic checkpoint sync
+
+To begin automatic checkpoint sync you will need HTTP API access to another synced beacon node.
+In this case, the URL could look like this: `http://remote-bn:5052`.
+
+> **Security Note**: You should cross-reference the `block_root` and `slot` of the loaded checkpoint
+> against a trusted source like a friend's node, or a block explorer.
```
Let's delete this section from here and move the checkpoint sync security check to the `Check logs` section.

michaelsproul commented on book/src/run_a_node.md:
```diff
@@ -0,0 +1,169 @@
+
+- [Geth: Connecting to Consensus Clients](https://geth.ethereum.org/docs/interface/consensus-clients)
+- [Nethermind: Running Nethermind & CL](https://docs.nethermind.io/nethermind/first-steps-with-nethermind/running-nethermind-post-merge)
+- [Besu: Connect to Mainnet](https://besu.hyperledger.org/en/stable/public-networks/get-started/connect/mainnet/)
```
Let's add Erigon to this section (and the merge migration docs). I think the go-to URL is https://github.com/ledgerwatch/erigon#beacon-chain-consensus-layer

michaelsproul commented on book/src/run_a_node.md:
````diff
@@ -0,0 +1,169 @@
+  --http
+```
+
+A Lighthouse beacon node can be configured to expose an HTTP server by supplying the `--http` flag. The default listen address is `127.0.0.1:5052`.
````
Let's add a sentence here about why the HTTP flag is needed:

> The HTTP API is required for the beacon node to accept connections from the _validator _client_, which manages keys.

michaelsproul commented on book/src/run_a_node.md:
```diff
@@ -0,0 +1,169 @@
+
+Once backfill is complete, a `INFO Historical block download complete` log will be emitted.
+
+Check out the [FAQ](./checkpoint-sync.md#faq) for more information on the checkpoint sync.
```
```suggestion
Check out the [FAQ](./checkpoint-sync.md#faq) for more information on checkpoint sync.
```

michaelsproul commented on book/src/run_a_node.md:
```diff
@@ -0,0 +1,169 @@
+This document provides detail for users who want to run a Lighthouse beacon node.
+You should be finished with one [Installation](./installation.md) of your choice to continue with the following steps:
+
+1. Enable [checkpoint sync](#step-1-checkpoint-sync);
+1. Set up an [execution node](#step-2-set-up-an-execution-node);
```
I'm torn as to whether we should swap the checkpoint sync and execution node sections, seeing as we talk about copying a URL to the clipboard.

michaelsproul commented on book/src/run_a_node.md:
````diff
@@ -0,0 +1,169 @@
+
+```
+INFO Loaded checkpoint block and state       state_root: 0xe8252c68784a8d5cc7e5429b0e95747032dd1dcee0d1dc9bdaf6380bf90bc8a6, block_root: 0x5508a20147299b1a7fe9dbea1a8b3bf979f74c52e7242039bd77cbff62c0695a, slot: 2034720, service: beacon
+```
````
This is where we should add a note about checking the checkpoint against one of the providers from https://eth-clients.github.io/checkpoint-sync-endpoints/, with another aside about "or a friend's node/block explorer"

michaelsproul reviewed (commented):
This is really great, thank you :clap:

I'm excited to get this merged!

kevinbogner committed:
Apply suggestions from code review

Co-authored-by: Michael Sproul <micsproul@gmail.com>

kevinbogner committed:
erigon, security note to step4, testnets, http api

kevinbogner committed:
switch step 1 with step 2

kevinbogner committed:
change URL

kevinbogner commented:
Thanks :) I made some changes:
- Committed your suggestions
- Added Erigon to 'Run a Node' and 'Merge Migration' section
- Added the security note to check against a trusted source to the 'Check logs' section
- Changed testnet to Goerli and added the testnets for Sepolia and Gnosis chain. 
- Switched step 1 with step 2. I think it is more straightforward now. I also deleted the 'clipboard sentence'. It was not ideal.

Let me know if there's anything else we should change.

Also saw a few checks failed. Maybe you can look into them and help me to understand what caused them. https://github.com/sigp/lighthouse/actions/runs/3444425151/jobs/5952589728 should be fixed now.

michaelsproul commented on book/src/run_a_node.md:
```diff
@@ -0,0 +1,171 @@
+# Run a Node
+
+This document provides detail for users who want to run a Lighthouse beacon node.
+You should be finished with one [Installation](./installation.md) of your choice to continue with the following steps:
```
```suggestion
You should be finished with one [Installation](./installation.md) method of your choice to continue with the following steps:
```

michaelsproul commented on book/src/run_a_node.md:
```diff
@@ -0,0 +1,171 @@
+1. [Further readings](#step-5-further-readings).
+
+Checkpoint sync is *optional*; however, we recommend it since it is substantially faster
+than syncing from genesis while still providing all the same features.
```
```suggestion
than syncing from genesis while still providing the same functionality.
```

michaelsproul reviewed (approved):
Approved modulo two minor tweaks to wording.

I'll merge now!

michaelsproul committed:
Last round of review

michaelsproul commented:
bors r+
